src/features/import-map.js

```diff
@@ -45,7 +45,15 @@ systemJSPrototype.resolve = function (id, parentUrl) {
 };
 
 function throwUnresolved (id, parentUrl) {
-  throw Error("Unable to resolve specifier '" + id + (parentUrl ? "' from " + parentUrl : "'"));
+  throw Error(errMsg(2, DEV ? "Unable to resolve specifier '" + id + (parentUrl ? "' from " + parentUrl : "'") : [id, parentUrl]));
+}
+
+function parseJson(script) {
+  try {
+    return JSON.parse(script);
+  } catch (err) {
+    throw Error(errMsg(1, DEV && "systemjs-import map contains invalid JSON"));
+  }
```

Comment:
Ok, here's a random idea for encapsulating this error system to one more level :)

```js
const parseJson = createCodeError(JSON.parse, DEV && 'SystemJS import map contains invalid JSON');
```

Where we use a function like the following:

```js
// global error number counter
let errCnt = 0;
function createCodeError(fn, msg) {
  const errCode = ++errCnt;
  return function () {
    try {
      fn.apply(this, arguments);
    }
    catch (e) {
      e.message = (msg || e.message) + ' - ' +
        'https://github.com/systemjs/systemjs/docs/errors.md#' + errCode;
      throw e;
  };
}
```

The benefit of this approach is not needing to explicitly define error codes since the "factory" does this for us.

Although they might go out of sync this way which is a risk too.

Just an idea, maybe food for thought.

Reply:
I like the idea, but don't think it would work, since extras are optionally included which would throw off the numbering.

Reply:
Yes, agreed the numbering should remain immutable and fully monotonic as the project changes. If we deprecate codes in future, we should just move them to a "deprecated" section where they can continue to be linkable etc etc.